apps/studio/components/interfaces/Database/Extensions/EnableExtensionModal.tsx

```diff
@@ -17,12 +19,23 @@ const EnableExtensionModal = ({ visible, extension, onCancel }: EnableExtensionM
   const { project } = useProjectContext()
   const { ui, meta } = useStore()
   const [defaultSchema, setDefaultSchema] = useState()
+  const [isEnabling, setIsEnabling] = useState(false)
```

Comment:
yup its because of the onSubmit method on line 74 - we're doing something else thats async prior to enabling the extension, hence using an external state to store the loading state

Reply:
tbh i was thinking whether this create schema if not exists statement should be within the react query for enabling extensions

Reply:
hmm actually yeah lemme do that